test/test_habitat_sim.py

```diff
@@ -82,3 +82,28 @@ def test_sim_no_sensors():
     sim = make_sim(config.SIMULATOR.TYPE, config=config.SIMULATOR)
     sim.reset()
     sim.close()
+
+
+def test_sim_geodesic_distance():
+    config = get_config()
+    if not os.path.exists(config.SIMULATOR.SCENE):
+        pytest.skip("Please download Habitat test data to data folder.")
+    sim = make_sim(config.SIMULATOR.TYPE, config=config.SIMULATOR)
+    sim.seed(0)
+    sim.reset()
+    navigable_points = [sim.sample_navigable_point() for _ in range(10)]
+    assert np.isclose(
+        sim.geodesic_distance(navigable_points[0], navigable_points[1]),
+        1.3849650,
+    ), "Geodesic distance or sample navigable points mechanism has been changed."
+    assert np.isclose(
+        sim.geodesic_distance(navigable_points[0], navigable_points[1:]),
+        1.3849650,
+    ), "Geodesic distance or sample navigable points mechanism has been changed."
+    assert sim.geodesic_distance(
```

Comment:
@erikwijmans, this test is failing and maybe surfacing an issue in MultiGoal calculations or current usage of the functionality:
```
E       AssertionError: Geodesic distance for multi target setup isn't equal to separate single target calls.
E       assert 1.3849650621414185 == [0.6194838285446167]
```

Reply:
Interesting that it is off by that much.  Unfortunately the multi-goal version won't be perfectly accurate as the closest target is selected in an intermediate step and before things are completely optimized, so it can select the wrong one.  In the little bit of testing I did, it was never off by that much, but I may have overfit to my scene/test cases. Guess we should use the version that is slower but exact (do a full dijkstra expand, get all paths, run the min over that, it will still be faster than running the N calls to A*, but slower than the single A* call done currently).

Reply:
Yes, to use this function reliably make sense to do it slower, but more accurate. For now I will comment out the last test and leave it for you to enable, once MultiGoal evaluation will be changed.

Reply:
Changed the multigoal to min(A*) in this PR: https://github.com/facebookresearch/habitat-sim/pull/465

Same functionality as you have here, but considerably faster as it's pushed all the way down in the stack.